### PR TITLE
infrastructure/checkers: Fill contents

### PR DESCRIPTION
**Pull request by razvand**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [ ] Tested your changes against relevant architectures and platforms;
- [ ] Updated relevant documentation (if needed).

## Description of changes

Update contents for "Validators and Checkers" section to the "Support Infrastructure" chapter.

### CODE REVIEW

File: chapters/infrastructure/checkers/reading/read.md [hunk 1]
```diff
@@ -1,12 +1,129 @@
 # Validators and Checkers
 
-Owner: Sergiu Weisz
-Reviewer: Mihai Băruța
+One of the greatest challenges of educators is the evaluation of learners submissions.
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:3]:
```suggestion
One of the greatest challenges of educators is the evaluation of learners' submissions.
```

File: chapters/infrastructure/checkers/reading/read.md [hunk 2]
```diff
@@ -1,12 +1,129 @@
...
+
+As resource usage increases, and demand is increased, `vmchecker` is able to **scale the amount of submissions that it checks in parallel**.
+A parallel approach means learners don't have to wait in long queues for assignment checking.
+
+`vmchecker` is **open source software** and is compatible with the [Open Educational Resources requirements](https://oercommons.org/).
+Using open source solutions allows us to quickly discover issues and make updates.
+
+Many education institutions use e-learning platforms to manage classrooms.
+These tools allow users to upload materials, evaluate learners, and grade assignments using a single application.
+`vmchecker` is integrated with existing educational platforms (such as Moodle) to it more appealing to integrate for institutions that already rely on these systems.
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:34]:
```suggestion
`vmchecker` is integrated with existing educational platforms (such as Moodle).
This makes it more appealing to institutions that already rely on these systems.
```

File: chapters/infrastructure/checkers/reading/read.md [hunk 3]
```diff
@@ -1,12 +1,129 @@
...
+* Learners make use of the submission repositories and the `vmchecker` installation to validate submissions and receive grading.
+
+Each role interacts with `vmchecker` through a user-friendly interface.
+This is the motivation for using GitLab that already provides the interface adapted for `vmchecker` scenarios.
+
+## Infrastructure Manager
+
+The infrastructure manager first needs to set up the `vmchecker` frontend and middleware.
+The [`vmchecker` Moodle plugin](https://github.com/open-education-hub/vmchecker-next-ui) is used as a frontend that can be copied and installed on the Moodle instance.
+The middleware is to be set up with all its required services using the Docker containers provided in the [`vmchecke` repository](https://github.com/open-education-hub/vmchecker-next).
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:71]:
```suggestion
The middleware is to be set up with all its required services using the Docker containers provided in the [`vmchecker` repository](https://github.com/open-education-hub/vmchecker-next).
```

File: chapters/infrastructure/checkers/reading/read.md [hunk 4]
```diff
@@ -1,12 +1,129 @@
...
+
+## Educator
+
+Educators need only interact with the frontend and GitLab to create a submission repository.
+The teacher must create a repository based on [the template provided by the `vmchecker` team](https://github.com/open-education-hub/vmchecker-next-assignment).
+The cloned Git repository will contain a GitLab CI/CD workflow file which defines the container which will run the submission and the script to run the assignment.
+The assignment repository must also contain a `src/` directory, which defines the directory in which students will write code.
+The Git repository must be private, otherwise students would be able to view the code amongst themselves.
+
+If using Moodle, educator must create an `Assignment` instance and a `vmchecker` assignment block, which will attach to the `vmchecker` backend.
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:87]:
```suggestion
If using Moodle, the educator must create an `Assignment` instance and a `vmchecker` assignment block, which will attach to the `vmchecker` backend.
```

File: chapters/infrastructure/checkers/reading/read.md [hunk 5]
```diff
@@ -1,12 +1,129 @@
...
+
+The container infrastructure presents an easier approach than the one seen in current solutions, which either require an internet connection for remote checking, or a virtual machine, which uses more resources.
+Using containers allows for easier patches to the environment, for example installing a new library, because the changes will be sent incrementally, the student not needing to download a whole new image.
+
+## Example `vmchecker` Deployments
+
+`vmchecker` has been used extensively for the Operating Systems class at the [National University of Science and Technology POLITEHNICA of Bucharest (UNSTPB)](https://upb.ro/), using the contents of [the `operating-systems` repository](https://github.com/open-education-hub/operating-systems).
+The repositories used for the five assignments, plus a template repository, plus a tutorial assignment can be used as examples of repositories for `vmchecker` deployment:
+
+- [`Assignment: Base`](https://gitlab.cs.pub.ro/operating-systems/assignment-base)
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:123]: This link returns a 404.

File: chapters/infrastructure/checkers/reading/read.md [hunk 6]
```diff
@@ -1,12 +1,129 @@
...
+- [`Assignment: Tutorial`](https://gitlab.cs.pub.ro/operating-systems/assignment-tutorial)
+- [`Assignment: Mini Libc`](https://gitlab.cs.pub.ro/operating-systems/assignment-mini-libc)
+- [`Assignment: Memory Allocator`](https://gitlab.cs.pub.ro/operating-systems/assignment-memory-allocator)
+- [`Assignment: Parallel Graph`](https://gitlab.cs.pub.ro/operating-systems/assignment-parallel-graph)
+- [`Assignment: Mini Shell`](https://gitlab.cs.pub.ro/operating-systems/assignment-mini-shell)
+- [`Assignment: Async Web Server`](https://gitlab.cs.pub.ro/operating-systems/assignment-async-web-server)
```
Comment (teodutu) [inline chapters/infrastructure/checkers/reading/read.md:129]: I think these links are pretty pointless overall. They are inaccessible without logging in using the UPB credentials, so only people from UPB will be able to access them.

**Review by teodutu — CHANGES_REQUESTED** [review]

**Comment by razvand** [comment]
@teodutu , links are publicly available: https://gitlab.cs.pub.ro/operating-systems/assignment-async-web-server

**Review by teodutu — APPROVED** [review]

**Pull request merged**